frontend/packages/noobaa-storage-plugin/src/components/status-card/utils.ts

```diff
@@ -0,0 +1,78 @@
+import * as _ from 'lodash';
+import { HealthState } from '@console/shared/src/components/dashboard/health-card/states';
+import { PrometheusResponse } from '@console/internal/components/graphs';
+import { FirehoseResult } from '@console/internal/components/utils';
+import { getGaugeValue } from '../../utils';
+
+export const getObjectStorageState: GetState = (results, error, isLoading) => {
+  const buckets = getGaugeValue(results[0]);
+  const unhealthyBuckets = getGaugeValue(results[1]);
+  const pools = getGaugeValue(results[2]);
+  const unhealthyPools = getGaugeValue(results[3]);
+  const noobaa = results[4];
+  const noobaaPhase = _.get(noobaa, 'data[0].status.phase');
+
+  if (error) {
+    return { state: HealthState.UNKNOWN };
+  }
+  if (isLoading) {
+    return { state: HealthState.LOADING };
+  }
+
+  if (noobaaPhase !== 'Ready') {
+    return {
+      state: HealthState.ERROR,
+      message: 'MCG is not running',
+    };
+  }
+  if (!_.isNil(pools) && !_.isNil(unhealthyPools)) {
+    if (Number(pools) === Number(unhealthyPools)) {
+      return {
+        state: HealthState.ERROR,
+        message: 'All resources are unhealthy',
+      };
+    }
+  }
+  if (!_.isNil(buckets) && !_.isNil(unhealthyBuckets)) {
+    if (Number(unhealthyBuckets) === 1) {
+      return {
+        state: HealthState.WARNING,
+        message: 'Object Bucket has an issue',
+      };
+    }
+    const value = unhealthyBuckets / buckets;
+    if (value >= 0.5) {
+      return {
+        state: HealthState.ERROR,
+        message: 'Many buckets have issues',
+      };
+    }
+    if (value >= 0.3) {
+      return {
+        state: HealthState.WARNING,
+        message: 'Some buckets have issues',
+      };
+    }
+  }
+  return { state: HealthState.OK };
+};
+
+export const getDataResiliencyState: GetState = (results, error) => {
+  const rebuildProgress = getGaugeValue(results[0]);
+  if (error) {
+    return { state: HealthState.UNKNOWN };
+  }
+  if (!results[0]) {
+    return { state: HealthState.LOADING };
+  }
+  if (rebuildProgress < 100) {
+    return { state: HealthState.PROGRESS };
+  }
+  return { state: HealthState.OK };
+};
+
+type GetState = (
+  results: (PrometheusResponse | FirehoseResult)[],
```

Comment:
Not in this case: https://github.com/openshift/console/pull/2918/files#diff-69ccfe9e1c84ad435346bff867d2171eR60

Reply:
I think she has tried making it generic for both the function

Reply:
Hence the comment in https://github.com/openshift/console/pull/2918/files#diff-69ccfe9e1c84ad435346bff867d2171eR60, to have two different types for the two different functions. Because one of the function is not meant to be used with FirehoseResult whereas the other one strictly needs both FirehoseResult and PrometheusResponse.

Reply:
I agree both should be separated

Reply:
Okay I will make separate definitions.

Reply:
This already does not applies after this [comment](https://github.com/openshift/console/pull/2918#discussion_r334330799).

I don't understand the point of conversation.
Guys, you should look at all the possibilities before suggesting